source/server/guarddog_impl.cc

```diff
@@ -84,11 +84,9 @@ GuardDogImpl::GuardDogImpl(Stats::Scope& stats_scope, const Server::Configuratio
 GuardDogImpl::~GuardDogImpl() { stop(); }
 
 void GuardDogImpl::step() {
-  {
-    Thread::LockGuard guard(mutex_);
-    if (!run_thread_) {
-      return;
-    }
+  Thread::LockGuard guard(mutex_);
+  if (!run_thread_) {
+    return;
```

Comment:
It's not clear to me why this lock now needs to be held for the entire function. Can you possibly add some more comments? Should there by GUARDED_BY annotations to make this more clear? Maybe this is just to simplify the logic?

Reply:
Added a comment.  This is part of the fix to test interlock; we need a full run of the step function after forceCheckForTest.  Performance shouldn't suffer since the use of mutex_ in the step() function should never be contended; the other uses of this mutex are GuardDogImpl::start, stop and forceCheckForTest.